src/actions/messages.js

```diff
@@ -29,53 +29,61 @@ let popping = false;
  */
 export function messageReceive(message) {
     return (dispatch, getState) => {
-        let messages = getState().messages.messages;
+        let { messages, messageQueue } = getState().messages;
         let {
-            delay,
+            active,
+            baseDelay,
             variance,
-            varianceMethod,
-            active
+            letterDelay,
+            minDelay,
+            maxDelay
         } = getState().config.typingStatus;
         let queueDelay = 0;
 
-        switch (varianceMethod) {
-            case 'fixed':
-                queueDelay = Math.round(
-                    delay + variance * Math.round(1.5 + Math.random() * -3)
-                );
-                break;
-            case 'range':
-            default:
-                queueDelay = Math.round(
-                    delay + variance * (1 + Math.random() * -2)
-                );
+        if (active) {
+            // Base delay plus the variance.
+            queueDelay = Math.round(
+                baseDelay + variance * Math.round(1.5 + Math.random() * -3)
+            );
+            // Add Delay per letter in the message.
+            let textSize = 0;
+            message.pages.map(page => {
+                textSize +=
+                    (page.title ? page.title.length : 0) +
+                    (page.text ? page.text.length : 0);
+            });
+            queueDelay += textSize * 40;
+            // Clamp to min and max delay size.
+            queueDelay = Math.max(Math.min(queueDelay, maxDelay), minDelay);
         }
 
         // Catch following messages or add first message.
         if (
             messageQueue.length === 0 &&
+            !waitingForMessage &&
             (messages.length === 0 ||
                 Date.now() - messages[messages.length - 1].timeAdded >
                     queueDelay)
         ) {
             dispatch(messageAdd(message));
         } else {
+            waitingForMessage = false;
             dispatch({
                 type: MESSAGE_QUEUE,
-                payload: message
+                payload: {
+                    message: message,
+                    delay: () =>
```

Comment:
Try to avoid storing functions in redux state. As a rule of thumb your state should always be serialisable. 
https://redux.js.org/docs/faq/OrganizingState.html#organizing-state-non-serializable

Reply:
Noted, removed and only storing the delay value and message

Reply:
^, pull the logic for your function out of this action and rather just store the config for the action?

```
payload: { 
  message,
  queueDelay
}

const fizz({message, queueDelay}) => {
  new Promise(resolve =>
    setTimeout(() => {
        dispatch(messageAdd(message));
            resolve();
                }, queueDelay)
            )
        }
}
```

or something like that...